lms/static/scripts/frontend_apps/components/OAuth2RedirectErrorApp.js

```diff
@@ -23,20 +23,32 @@ export default function OAuth2RedirectErrorApp({ location = window.location }) {
     OAuth2RedirectError: {
       authUrl = /** @type {string|null} */ (null),
       invalidScope = false,
+      errorCode = '',
```

Comment:
Adding a new errorCode attribute here instead of a new "missingBBIntegration" or similar. I think we'll have an big proliferation of booleans here otherwise. If you like this approach I'll follow up with another PR to refactor "invalidScope" as an errorCode.

Reply:
:+1: This is _definitely_ better than a boolean per possible thing that could go wrong.